### ISSUE TITLE
proper way to pad prompts

### ISSUE THREAD

**Issue by HassanJbara** [jzhang38/TinyLlama#159]
I'm trying to fine-tune the model and for that I need to pad the dataset prompts. However, when I se left padding with the eos token, the outputs devolve into gibberish. Am I doing padding wrong? What is the solution here?

![image](https://github.com/jzhang38/TinyLlama/assets/82707653/35d37320-d908-4bc0-81a4-dbbba52c03e2)

I'm using `TinyLlama-1.1B-Chat-v1.0`

**Comment by jzhang38** [comment]
https://github.com/OpenAccess-AI-Collective/axolotl/tree/main/examples/tiny-llama

You can have a look at this fine-tuning framework where they provide examples for tinyllama.

**Comment by HassanJbara** [comment]
> https://github.com/OpenAccess-AI-Collective/axolotl/tree/main/examples/tiny-llama
> 
> You can have a look at this fine-tuning framework where they provide examples for tinyllama.

I now realize that the problem was not passing the attention mask generated by the tokenizer. I have seen models with designated padding tokens who work just fine without attention masks, but I guess llama models need it.
